src/hooks/http.ts

```diff
@@ -264,32 +264,39 @@ export class Http {
     response
   ) {
     let body = '';
-    const payloadSize = getEventEntitySize();
+    let payloadSize = getEventEntitySize(true);
     return function (args) {
       GlobalDurationTimer.start();
       const receivedTime = new Date().getTime();
+      let truncated = false;
       const { headers, statusCode } = response;
       if (args[0] === 'data' && body.length < payloadSize) {
         let chunk = args[1].toString();
         const allowedLengthToAdd = payloadSize - body.length;
         if (chunk.length > allowedLengthToAdd) {
+          truncated = true;
           chunk = chunk.substr(0, allowedLengthToAdd);
         }
         body += chunk;
       }
       if (args[0] === 'end') {
+        let maxSizeNoErrors = getEventEntitySize();
         const responseData = {
           statusCode,
           receivedTime,
-          body,
+          body:
+            statusCode < 400 && body.length > maxSizeNoErrors
+              ? body.substr(0, maxSizeNoErrors)
+              : body,
```

Comment:
Are we taking the all body in this case? even if it's 1GB?

Reply:
no there is a max of 2048KB with no error and 4096KB with errors

Reply:
if (chunk.length > allowedLengthToAdd) { 

makes sure of that